### PR TITLE
[Merged by Bors] - feat: add `PowerBasis.exists_smodEq`

### PR DESCRIPTION
**Pull request by riccardobrasca**
We add `PowerBasis.exists_sModEq`.

From `flt-regular`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by riccardobrasca** [comment]
!bench

File: Mathlib/RingTheory/PowerBasis.lean [hunk 1]
```diff
@@ -143,6 +144,25 @@ theorem algHom_ext {S' : Type*} [Semiring S'] [Algebra R S'] (pb : PowerBasis R
   rw [← Polynomial.aeval_algHom_apply, ← Polynomial.aeval_algHom_apply, h]
 #align power_basis.alg_hom_ext PowerBasis.algHom_ext
 
+open Ideal Finset Submodule in
+theorem exists_sModEq (pb : PowerBasis A B) (b : B) :
```
Comment (erdOne) [inline Mathlib/RingTheory/PowerBasis.lean:148]: Can you provide a version with `∃ a, pb.gen ∣ b - algebraMap A B a`?

Comment (riccardobrasca) [reply Mathlib/RingTheory/PowerBasis.lean:148]: Done, thanks!

File: Mathlib/RingTheory/PowerBasis.lean [hunk 2]
```diff
@@ -143,6 +144,28 @@ theorem algHom_ext {S' : Type*} [Semiring S'] [Algebra R S'] (pb : PowerBasis R
...
```
Comment (eric-wieser) [inline Mathlib/RingTheory/PowerBasis.lean:148]: I think the naming convention is
```suggestion
theorem exists_smodEq (pb : PowerBasis A B) (b : B) :
```
judging by how `zmodEq` is used.

File: Mathlib/RingTheory/PowerBasis.lean [hunk 3]
```diff
@@ -143,6 +144,28 @@ theorem algHom_ext {S' : Type*} [Semiring S'] [Algebra R S'] (pb : PowerBasis R
...
+  · rw [Quotient.mk_zero, Quotient.mk_eq_zero, coe_basis]
+    refine sum_mem _ (fun i hi ↦ ?_)
+    rw [Algebra.smul_def']
+    refine Ideal.mul_mem_left _ _ <| Ideal.pow_mem_of_mem _ (Ideal.subset_span (by simp)) _ <|
+      Nat.pos_of_ne_zero <| fun h ↦ not_mem_erase i univ <| Fin.eq_mk_iff_val_eq.2 h ▸ hi
+
+open Submodule.Quotient in
+theorem exists_gen_dvd_sub (pb : PowerBasis A B) (b : B) : ∃ a, pb.gen ∣ b - algebraMap A B a := by
+  simpa [← Ideal.mem_span_singleton, ← mk_eq_zero, mk_sub, sub_eq_zero] using pb.exists_smodEq b
 section minpoly
```
Comment (erdOne) [inline Mathlib/RingTheory/PowerBasis.lean:169]:
```suggestion

section minpoly
```

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**